salsah/src/public/js/jquery.searchbox.js

```diff
@@ -206,7 +208,12 @@
 							left: pos.left,
 							top: pos.top
 						}).show();
-						$this.val('');
+						// if the input already has a value and it is valid, back it up
```

Comment:
if the input box never had focus, it holds no value, there is then no need for a cache.  
The problem is that we want to clear an existing text to let room for the user to type in a new text to search, but this new text search might not end up in a valid answer.  
Still the input value is the text the user typed in so far.

That's why I added a `isValid`, if we find a value and click on its line in the drop-down list, then we set the input's value with a new text and we set the `isValid` attribute to true.

But by default we have to set it to false.

As it is a html tag, it gets turned into a string... it might help clarity if I put all `isValid` assignment to string?